lib/signaling/v2/peerconnection.js

```diff
@@ -718,6 +732,9 @@ class PeerConnectionV2 extends StateMachine {
       if (isFirefox) {
         description.sdp = filterOutMediaStreamIds(description.sdp);
       }
+      if (!this._peerConnection.remoteDescription) {
+        this._isIceLite = /a=ice-lite/.test(description.sdp);
```

Comment:
@makarandp0 

> 1. you are setting this._isIceLite only if remoteDescription is not already set - is it because 2nd setRemoteDescription would get rejected anyways?

Determining whether the remote peer is an ICE-lite agent is a one time operation per PeerConnection. So we only calculate it when we receive the first remote RTCSessionDescription.

> 2. You are setting this._isIceLite before this._peerConnection.setRemoteDescription resolves. if it rejects then we would we end-up with this._isIceLite set to invalid value?

`_iceLite` should be calculated only once, irrespective of whether `setRemoteDescription` resolves or rejects because its calculation is not dependent on whether the SDP is valid or not. It is only dependent on whether `a=ice-lite` is present in the SDP.

Reply:
makes sense. Thanks